### PR TITLE
[openhabcloud] improved logging for troubleshooting purposes

### PR DESCRIPTION
**Pull request by ssalonen**
Improved logging for openhabcloud addon:
- Sensor sensitive secrets from logging statements
- log error reason even without debug level
- more diagnostic logging, e.g. Socket.io ping/pong and HTTP calls

In addition,
- removed dead code (was included in the original contribution from @kaikreuzer ) https://github.com/openhab/openhab-addons/commit/f85d2bbaaec89848ee05d8c638e70d45bbca5858
- abort correctly on case of invalid URL for example (also longstanding bug?) https://github.com/openhab/openhab-addons/commit/885aacea5929093d0fd43b727c2430a643bff822

https://community.openhab.org/t/openhab-cloud-connection-stops-working-does-not-try-to-reconnect/133608/17

**Precompiled versions** for openHAB 3.2.0 and 3.3.0-SNAPSHOT: https://github.com/ssalonen/openhab2-addons/releases?q=openhabcloud&expanded=true Also available as openHAB marketplace addons

### CODE REVIEW

**Comment by lolodomo** [comment]
Can you please provide a better title?

**Comment by ssalonen** [comment]
Ready for review.

Asking @kaikreuzer to double check commits highlighted in the description.

File: bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java [hunk 1]
```diff
@@ -684,4 +694,11 @@ private void sleepSocketIO(long delay) {
             }
         });
     }
+
+    private static String sensored(String secret) {
```
Comment (Flole998) [inline bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java:698]: I think you mean censored?

File: bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudService.java [hunk 2]
```diff
@@ -311,16 +311,23 @@ private String getSecret() {
...
         } else {
             newSecretString = readFirstLine(file);
-            logger.debug("Using secret at '{}' with content '{}'", file.getAbsolutePath(), newSecretString);
+            logger.debug("Using secret at '{}' with content '{}'", file.getAbsolutePath(), sensored(newSecretString));
         }
 
         return newSecretString;
     }
 
+    private static String sensored(String secret) {
```
Comment (Flole998) [inline bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudService.java:324]: Same thing here

Comment (ssalonen) [reply bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudService.java]: https://github.com/openhab/openhab-addons/pull/12430/commits/fc8043adb2d62c13664503d49ac7e5c0910c6aac

File: bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java (hunk 1 above)
Comment (ssalonen) [reply bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: https://github.com/openhab/openhab-addons/pull/12430/commits/fc8043adb2d62c13664503d49ac7e5c0910c6aac

**Comment by lolodomo** [comment]
Is it still in progress?

**Review by Flole998 — APPROVED** [review]
LGTM

File: bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java [hunk 3]
```diff
@@ -157,159 +163,170 @@ public CloudClient(HttpClient httpClient, String uuid, String secret, String bas
...
-                        logger.error("Error during communication: {}", args[0]);
+                })//
+                .on(Socket.EVENT_RECONNECT_FAILED,
+                        args -> logger.debug("Socket.IO re-connect attempts failed. Stopping reconnection."))//
+                .on(Socket.EVENT_DISCONNECT, args -> {
+                    if (args.length > 0) {
+                        logger.info("Socket.IO disconnected: {}", args[0]);
                     } else {
-                        logger.error("Error during communication");
+                        logger.info("Socket.IO disconnected");
```
Comment (fwolter) [inline bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java:270]: Shouldn't these be debug (recoverable error) or warn?

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java (hunk 3 above)
Comment (ssalonen) [reply bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: Made it warn in https://github.com/openhab/openhab-addons/pull/12430/commits/49cec9cf3d16454d64c14bbfde5e879e2fb7bd01

I think it's better that by default users see when disconnects happen

**Review by fwolter — APPROVED** [review]

**Pull request merged**